### PR TITLE
Allow tab character before first annotation in DocBlock

### PR DESCRIPTION
**Pull request by poldridge**
https://github.com/doctrine/annotations/issues/68

### CODE REVIEW

**Comment by kimhemsoe** [comment]
Could you add a tests to go this?

**Comment by poldridge** [comment]
Okay, I added a test.

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 1]
```diff
@@ -1239,6 +1239,21 @@ public function testTrailingCommaIsAllowed()
...
+    {
+        $docblock = <<<DOCBLOCK
+/**
+ *	@Name
+ */
+DOCBLOCK;
+
+        $parser = $this->createTestParser();
+        $result = $parser->parse($docblock);
+        $this->assertEquals(1, count($result));
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: `assertCount(1, $result)`

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 2]
```diff
@@ -1239,6 +1239,21 @@ public function testTrailingCommaIsAllowed()
...
+        $annot = $result[0];
+        $this->assertTrue($annot instanceof Name);
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: `self::assertInstanceOf('Name', reset($result))`

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 3]
```diff
@@ -348,7 +348,7 @@ private function findInitialTokenPosition($input)
         // search for first valid annotation
         while (($pos = strpos($input, '@', $pos)) !== false) {
             // if the @ is preceded by a space or * it is valid
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocParser.php]: Comment needs updating as well

**Comment by poldridge** [comment]
@Ocramius While I agree using self with the scope resolution operator is more correct than `$this` for accessing a static method, it isn't used in any of the other tests. As far as I can tell `$this->assertX` is used throughout all the tests, and `self::assert` is never used; so I kept the same style for consistency reasons.

**Comment by Ocramius** [comment]
@poldridge agree on the style, but at least you shouldn't `assertTrue($expression)`, and instead do `assertInstanceOf` there ;-)

**Comment by poldridge** [comment]
@Ocramius Okay, I'll update it to that.

Note though that `assertTrue($x instanceof Y)` occurs 25 times in DocParserTest.php alone. If there is consensus that `$this->assertTrue($expression);` statements should be replaced with the more semantic `self::assertX($var)`, perhaps a new issue should be created for updating that in every test.

**Comment by Ocramius** [comment]
@poldridge we'll probably do that during a cleanup when aiming at 2.x :-)

**Comment by alvinbunk** [comment]
I also noticed that for annotations within the comment block, if the 1st line of the comment block does not have "/***", and has "/**" instead, the annotation doesn't get recognized. I'm not sure if this is a problem?

For example, this works:

```
/**
 * @Assert\NotBlank()
 */
```

But this doesn't:

```
/*
 * @Assert\NotBlank()
 */
```

**Comment by Ocramius** [comment]
That is normal/expected behavior: without `/**`, it is just a comment, and php's reflection won't retrieve it anyway.

**Comment by alvinbunk** [comment]
Thank you for letting me know @Ocramius !

**Comment by Ocramius** [comment]
Moved to #99

**Pull request closed without merging**